src/hotspot/share/runtime/frame.cpp

```diff
@@ -885,6 +885,7 @@ oop frame::interpreter_callee_receiver(Symbol* signature) {
 void frame::oops_interpreted_do(OopClosure* f, const RegisterMap* map, bool query_oop_map_cache) const {
   assert(is_interpreted_frame(), "Not an interpreted frame");
   Thread *thread = Thread::current();
+  DEBUG_ONLY(ResourceMark rm(thread);) // ~InterpreterOopMap already handles possible deallocation of bitmask
```

Comment:
I don't like that this is in debug mode only either.  We have the current thread so that's a part of the cost of ResourceMark (or historically been the cost).  I wonder if this "optimization" can be observed.  I'd rather the explicit resource allocation deletion be removed also, since this is surprising.

Reply:
I'm guessing the explicit resource array deletion was an attempt to save memory if the ResourceMark was outside a loop iterating frames.  But if the ResourceMark is inside the loop, it seems pointless.  I think uses of other frame iterators like vframes have the ResourceMark on the outside, and that apparently hasn't caused memory footprint issues.